cmd/copy.go

```diff
@@ -816,180 +813,27 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		blobParts := azblob.NewBlobURLParts(*fromUrl)
-		cca.sourceSAS = blobParts.SAS.Encode()
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		blobParts.SAS = azblob.SASQueryParameters{}
-		bUrl := blobParts.URL()
-		cca.source = bUrl.String()
-
-		// set the clean source root
-		if strings.Contains(blobParts.ContainerName, "*") {
-			if blobParts.BlobName != "" {
-				return errors.New("cannot combine a wildcarded container name and blob path")
-			}
-
-			blobParts.ContainerName = ""
-		}
-		bUrl = blobParts.URL()
-		jobPartOrder.SourceRoot = bUrl.String()
-	case common.ELocation.File():
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		fileParts := azfile.NewFileURLParts(*fromUrl)
-		cca.sourceSAS = fileParts.SAS.Encode()
-		if cca.sourceSAS == "" {
-			return fmt.Errorf("azure files only supports SAS token authentication")
-		}
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		fileParts.SAS = azfile.SASQueryParameters{}
-		fUrl := fileParts.URL()
-		cca.source = fUrl.String()
-
-		// set the clean source root
-		if strings.Contains(fileParts.ShareName, "*") {
-			if fileParts.DirectoryOrFilePath != "" {
-				return errors.New("cannot combine a wildcarded share name and file path")
-			}
-
-			fileParts.ShareName = ""
-		}
-		fUrl = fileParts.URL()
-		jobPartOrder.SourceRoot = fUrl.String()
-
-	case common.ELocation.BlobFS():
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		bfsParts := azbfs.NewBfsURLParts(*fromUrl)
-		cca.sourceSAS = bfsParts.SAS.Encode()
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		bfsParts.SAS = azbfs.SASQueryParameters{}
-		bfsUrl := bfsParts.URL()
-		cca.source = bfsUrl.String() // this escapes spaces in the source
-
-		// set the clean source root
-		if strings.Contains(bfsParts.FileSystemName, "*") {
-			if bfsParts.DirectoryOrFilePath != "" {
-				return errors.New("cannot combine a wildcarded filesystem name and file path")
-			}
-
-			bfsParts.FileSystemName = ""
-		}
-		bfsUrl = bfsParts.URL()
-		jobPartOrder.SourceRoot = bfsUrl.String()
-
-	case common.ELocation.S3():
-		fromURL, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromURL.String(), err.Error())
-		}
-
-		// S3 management console encode ' '(space) as '+', which is not supported by Azure resources.
-		// To support URL from S3 managment console, azcopy decode '+' as ' '(space).
-		*fromURL = common.URLExtension{URL: *fromURL}.URLWithPlusDecodedInPath()
-		cca.source = fromURL.String()
+	cca.source, cca.sourceSAS, err = SplitAuthTokenFromResource(cca.source, from)
```

Comment:
What happened to the ToExtendedPath and cleanLocalPath calls that used to be here for local locations?  Have they gone?  Is that safe? (E.g. are they happening somewhere else anyway?)

Reply:
This change may well have been a massive mistake.

I for some reason, didn't think about the local case and went "ah, it's fine, all of this does the same thing, anyway."

Reply:
I'd like to maintain what I've done so I'll re-add the missing effects in one way or another.